mobile_config/src/mobile_radio_tracker.rs

```diff
@@ -208,8 +208,21 @@ async fn get_tracked_radios(
 fn get_all_mobile_radios(metadata: &Pool<Postgres>) -> impl Stream<Item = MobileRadio> + '_ {
     sqlx::query_as::<_, MobileRadio>(
         r#"
+        WITH unique_entity_keys AS (
+            SELECT
+                kta.entity_key as entity_key,
+                kta.asset as asset,
+                MAX(mhi.refreshed_at) AS refreshed_at
+            FROM key_to_assets kta
+            INNER JOIN mobile_hotspot_infos mhi
+                ON kta.asset = mhi.asset
+            WHERE kta.entity_key IS NOT NULL
+              AND mhi.refreshed_at IS NOT NULL
+            GROUP BY kta.entity_key, kta.asset
+        )
```

Comment:
can this query be simplified back into something more like the original with a `SELECT DISTINCT ON (entity_key, asset) ... ORDER BY refreshed_at DESC` to limit results to the most recent single entry for that given combo of identifiers?

Reply:
Thank you for the hint. Now it is much better! 

https://www.postgresql.org/docs/current/sql-select.html#SQL-DISTINCT